### ISSUE TITLE
Apple M1: bazel tests can't run due to dyld link error

### ISSUE THREAD

**Issue by devjgm** [googleapis/google-cloud-cpp#6443]
On a Mac mini with an M1 chip, I checked out this repo and installed bazel from homebrew `release: release 4.0.0-homebrew `

I was able to compile all the code

```
$ bazel build ...
```

But running the tests all failed:

```
$ bazel test --test_output=errors --test_tag_filters=-integration-test ...
```

The errors all look like:

```
==================== Test output for //google/cloud/storage:internal_resumable_upload_session_test:
dyld: Symbol not found: _aes_hw_cbc_encrypt
  Referenced from: /private/var/tmp/_bazel_jgm/06967dc09dcdceaf2c120ba0a41576e9/sandbox/darwin-sandbox/4286/execroot/com_github_googleapis_google_cloud_cpp/bazel-out/darwin-fastbuild/bin/google/cloud/storage/internal_resumable_upload_session_test.runfiles/com_github_googleapis_google_cloud_cpp/google/cloud/storage/../../../_solib_darwin//libexternal_Sboringssl_Slibcrypto.so
  Expected in: flat namespace
 in /private/var/tmp/_bazel_jgm/06967dc09dcdceaf2c120ba0a41576e9/sandbox/darwin-sandbox/4286/execroot/com_github_googleapis_google_cloud_cpp/bazel-out/darwin-fastbuild/bin/google/cloud/storage/internal_resumable_upload_session_test.runfiles/com_github_googleapis_google_cloud_cpp/google/cloud/storage/../../../_solib_darwin//libexternal_Sboringssl_Slibcrypto.so

```

**Comment by devjgm** [comment]
From @coryan offline: 
> FWIW, the Bazel builds use BoringSSL.  We get that indirectly through gRPC.  It may be worthwhile checking that gRPC is pulling a recent-enough version of BoringSSL with good M1 support.  It is also worthwhile checking if  OpenSSL or LibreSSL have better support for M1, and therefore and easier path to compile+test on M1

**Comment by devjgm** [comment]
It looks to me like BoringSSL does not have Apple ARM support. The BoringSSL source comes from here: https://github.com/google/boringssl/tree/main-with-bazel The needed functions are _declared_ here:

https://github.com/google/boringssl/blob/59667dfe33fb27fe835eb8800fc6396a60d5ee13/src/crypto/fipsmodule/aes/internal.h#L79-L87

However, it looks like their implementation should come form a platform/hardware specific location here: https://github.com/google/boringssl/tree/main-with-bazel likely named something like `mac-arm64`, which does not exist.

**Comment by devjgm** [comment]
BoringSSL issue filed: https://bugs.chromium.org/p/boringssl/issues/detail?id=411

**Comment by devjgm** [comment]
This is now fixed with bazel 4.1.
